inject_version.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+# set from Makefile
+VERSION="$(date +v%Y%m%d)-$(git describe --tags --always)"
```

Comment:
Why date at the start of the version? This would give older branches/commits/tags a "higher" version number if they are installed after a newer version.

Reply:
Fair enough. I can fix that.

Reply:
OK, so I've [been looking](https://git-scm.com/docs/git-describe) into `git describe` a bit as I haven't used it in ages and we don't typically use it in our shop. 
Help me understand how your comment, @sparr  is accurate with respect to how `describe` is currently used for this project. If it's not typically or currently used (as it seems is likely the case), please help me understand your comment preferably with samples relevant to this project so I can craft something appropriate for this project. From my testing, it seems that `git describe` always returns the latest commit which is relevant to the most release, which makes the date relevant since releases should be linear, right? The last set of commit(s) to the release should be irrelevant and this line is specifically relevant to releases. I look forward to your response.

Reply:
If you build 1.3.0-8-g648fe8e today you'll get the version number you just produced.

If I then build 1.2.1 next week, I'll produce a version number that looks like `v20230624-1.2.1` which most platforms will interpret as a higher version number.

I suggest that the default behavior should never produce a higher version number from an older version/tag/branch. If someone needs to cause that outcome, it should be intentional and exceptional.

Reply:
It is not common for our repo to build two separate minor versions regularly. The most plausible failure mode here would be us building a version to release and then a distro building an earlier version (such as to backport a security fix) later. Users attempting to still use the newest version would get downgraded.